pkg/kubelet/kubelet.go

```diff
@@ -1206,6 +1208,10 @@ type Kubelet struct {
 	// status to master. It is only used when node lease feature is enabled.
 	nodeStatusReportFrequency time.Duration
 
+	// next node status report interval after adding some randomness to the nodeStatusReportFrequency to
+	// help spread the load evenly over time.
+	nextNodeStatusReportInterval time.Duration
```

Comment:
I see, so `nodeStatusUpdateFrequency` is the one that has the dependencies on controllers and `nodeStatusReportFrequency` depends only on the leases ...

Reply:
why adding a new field here and not calculating it directly on `tryUpdateNodeStatus` loop?

Reply:
>why adding a new field here and not calculating it directly on tryUpdateNodeStatus loop?

Trying to ensure `nodeStatusReportFrequency` is bounded. IOW it is guaranteed that it won't be more than 125% of user specified `nodeStatusReportFrequency` value.

If we do `new nodeStatusReportFrequency = [75%, 125%] * old nodeStatusReportFrequency`, the worst case is that it keeps getting multiplied by 1.25 repeatedly and after long enough time, `nodeStatusReportFrequency` can become much significantly longer than the user specified `nodeStatusReportFrequency`. I'm trying to prevent this worst case scenario.